src/convertToObject.js

```diff
@@ -6,7 +6,22 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.trim()) {
+    return {};
+  }
+
+  return sourceString
+    .split(';')
+    .map((str) => str.trim())
+    .filter((str) => str)
+    .map((str) => str.split(':').map((part) => part.trim()))
```

Comment:
This line splits each string by a colon and trims the parts. However, if a string doesn't contain a colon, `str.split(':')` will return an array with a single element, which could lead to unexpected behavior in the `reduce` function. Consider adding validation to ensure each string contains a colon before splitting.